CMakeLists.txt

```diff
@@ -90,86 +85,53 @@ option(IREE_ENABLE_EMITC "Enables MLIR EmitC dependencies." ${IREE_BUILD_COMPILE
-
-# Start each IREE_HAL_DRIVER_* OFF, then enable from IREE_HAL_DRIVERS_TO_BUILD.
-foreach(_backend ${IREE_ALL_HAL_DRIVERS})
-  string(TOUPPER "${_backend}" uppercase_backend)
-  set(IREE_HAL_DRIVER_${uppercase_backend} OFF CACHE BOOL "" FORCE)
-endforeach()
-foreach(_backend ${IREE_HAL_DRIVERS_TO_BUILD})
-  string(TOUPPER "${_backend}" uppercase_backend)
-  string(REPLACE "\"" "" uppercase_backend ${uppercase_backend})
-  set(IREE_HAL_DRIVER_${uppercase_backend} ON CACHE BOOL "" FORCE)
-endforeach()
-
-# List of all compiler target backends included in the project:
-set(IREE_ALL_TARGET_BACKENDS
-  CUDA
-  Dylib-LLVM-AOT
-  WASM-LLVM-AOT
-  Metal-SPIRV
-  ROCm
-  Vulkan-SPIRV
-  VMVX
-  WebGPU
-)
 
-if(${IREE_BUILD_COMPILER})
-  if(IREE_TARGET_BACKENDS_TO_BUILD STREQUAL "default")
-    set(IREE_TARGET_BACKENDS_TO_BUILD ${IREE_ALL_TARGET_BACKENDS})
-
-    # Disable the WebGPU backend by default, as it has complex dependencies and
-    # is still early in its development.
-    list(REMOVE_ITEM IREE_TARGET_BACKENDS_TO_BUILD WebGPU)
-  endif()
-  message(STATUS "Building compiler target backends: ${IREE_TARGET_BACKENDS_TO_BUILD}")
+# Vulkan is not natively supported on Apple platforms.
+# Metal should generally be used instead, though MoltenVK may also work.
+if (APPLE)
+  set(IREE_HAL_DRIVER_VULKAN_DEFAULT OFF)
 else()
-  set(IREE_TARGET_BACKENDS_TO_BUILD "" CACHE STRING "" FORCE)
-  message(STATUS "Compiler is disabled, building no target backends")
+  set(IREE_HAL_DRIVER_VULKAN_DEFAULT ON)
 endif()
 
-# Default every IREE_TARGET_BACKEND_* to OFF
-foreach(_backend ${IREE_ALL_TARGET_BACKENDS})
-  string(TOUPPER "${_backend}" uppercase_backend)
-  set(IREE_TARGET_BACKEND_${uppercase_backend} OFF CACHE BOOL "" FORCE)
-endforeach()
-
-# Set IREE_TARGET_BACKEND_* based on configuration
-foreach(_backend ${IREE_TARGET_BACKENDS_TO_BUILD})
-  string(TOUPPER "${_backend}" uppercase_backend)
-  string(REPLACE "\"" "" uppercase_backend ${uppercase_backend})
-  set(IREE_TARGET_BACKEND_${uppercase_backend} ON CACHE BOOL "" FORCE)
-endforeach()
-
-list(APPEND CMAKE_MODULE_PATH
-  ${CMAKE_CURRENT_LIST_DIR}/build_tools/cmake/
-  ${CMAKE_CURRENT_LIST_DIR}/bindings/python/build_tools/cmake/
-)
+option(IREE_HAL_DRIVER_CUDA "Enables the CUDA runtime HAL driver" ${IREE_HAL_DRIVER_CUDA_DEFAULT})
+option(IREE_HAL_DRIVER_DYLIB "Enables the Dylib runtime HAL driver" ON)
+option(IREE_HAL_DRIVER_DYLIB_SYNC "Enables the Dylib_Sync runtime HAL driver" ON)
+option(IREE_HAL_DRIVER_VMVX "Enables the VMVX runtime HAL driver" ON)
+option(IREE_HAL_DRIVER_VMVX_SYNC "Enables the VMVX_Sync runtime HAL driver" ON)
+option(IREE_HAL_DRIVER_VULKAN "Enables the Vulkan runtime HAL driver" ${IREE_HAL_DRIVER_VULKAN_DEFAULT})
+
+option(IREE_TARGET_BACKEND_CUDA "Enables the CUDA compiler target backend" ${IREE_BUILD_COMPILER})
+option(IREE_TARGET_BACKEND_DYLIB-LLVM-AOT "Enables the Dylib-LLVM-AOT compiler target backend" ${IREE_BUILD_COMPILER})
```

Comment:
Can we not mix dashes and underscores please? I will get this wrong every single time